server/app/services/program/ProgramServiceImpl.java

```diff
@@ -705,6 +789,86 @@ public ProgramDefinition setBlockEligibilityDefinition(
     return updateProgramDefinitionWithBlockDefinition(programDefinition, blockDefinition);
   }
 
+  @Override
+  public ProgramDefinition setBlockEligibilityDefinition(
+      long programId, long blockDefinitionId, DynamicForm predicateForm)
```

Comment:
Services shouldn't be interacting with Forms as that is a FrontEnd concern which Controllers manage and Services should not handle.  

So this is pretty smelly, and I don't see a compelling reason to not do it in the Controller.

Reply:
done